### DISCUSSION TITLE
Best way to disable snippits?

### DISCUSSION THREAD

**Discussion by GNOMES** [VonHeikemen/lsp-zero.nvim#376] [Q&A]
I am not personally a fan of snippits.

Is there an LSP-zero suggested method to disable these? Otherwise I see people doing this from nvim-cmp.

**Comment by VonHeikemen** [comment]
What version of lsp-zero are you using?

**Comment by GNOMES** [comment]
I am following the 3.x branch

**Reply by VonHeikemen** [reply]
I've updated lsp-zero. I added a new option that should help with that.

If you set the global variable `lsp_zero_extend_capabilities` to `0` before using lsp-zero, then language servers should stop sending snippets in the code completion suggestions.

```lua
vim.g.lsp_zero_extend_capabilities = 0
```

Note this might not work for all language servers. They can still ignore the settings lsp-zero sends, and there is nothing I can do.

**Comment by GNOMES** [comment]
Would there still be a dependency on LuaSnip with this option enabled?

**Reply by VonHeikemen** [reply]
If you don't install luasnip nvim-cmp will work. But, there is no way to prevent nvim-cmp from trying to expand a snippet. So, if a language server sends a snippet and you select that, you'll get an error.

The best you can do is provide an "empty function" to nvim-cmp's expand function.

```lua
  local cmp = require('cmp')
  cmp.setup({
    snippet = {
      expand = lsp_zero.noop,
    },
  })
```

This will not raise an error but probably have other side effects.
